compiler/src/iree/compiler/Codegen/Common/MathTransformPass.cpp

```diff
@@ -23,8 +28,85 @@ static llvm::cl::opt<bool> clNativeMathPrecision(
                    "for math op natively available on GPU.\""),
     llvm::cl::init(false));
 
-#define GEN_PASS_DEF_MATHTRANSFORMPASS
-#include "iree/compiler/Codegen/Common/Passes.h.inc"
+static llvm::cl::opt<std::string> clMathTransformTweaks(
+    "iree-codegen-math-transform-tweaks",
+    llvm::cl::desc(
+        "Comma-separated list of tweaks to apply to default math transforms. "
+        "Each entry has the form +transform:op or -transform:op to "
+        "enable/disable a transform on an op. Allowed values for `op` are the "
+        "math dialect ops, e.g. math.cos. Allowed values for `transform` "
+        "are: rewrite, approx, f32cast. Example: -approx:math.cos."),
+    llvm::cl::init(""));
+
+struct MathTransformTweaks {
+  // Ops to enable "rewrite" transformations on. See `predicateRewrite`.
```

Comment:
I think we do not need the `See *` comment because it is asking users to look at implementation. There are no comments on those methods.

Instead, we can document what the string format is, and what happens if an op shows up in both vectors. I think `enable` one has higher priority.